toolchain/check/testdata/impl/fail_impl_bad_interface.carbon

```diff
@@ -8,19 +8,39 @@
 // TIP: To dump output, run:
 // TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/impl/fail_impl_bad_interface.carbon
 
-// CHECK:STDERR: fail_impl_bad_interface.carbon:[[@LINE+10]]:1: error: semantics TODO: `impl as non-interface` [SemanticsTodo]
-// CHECK:STDERR: impl i32 as false {}
-// CHECK:STDERR: ^~~~~~~~~~~~~~~~~~~
-// CHECK:STDERR:
-// CHECK:STDERR: fail_impl_bad_interface.carbon:[[@LINE+6]]:13: error: cannot implicitly convert from `bool` to `type` [ImplicitAsConversionFailure]
+// --- fail_impl_as_false.carbon
+
+library "[[@TEST_NAME]]";
+
+// CHECK:STDERR: fail_impl_as_false.carbon:[[@LINE+7]]:13: error: cannot implicitly convert from `bool` to `type` [ImplicitAsConversionFailure]
 // CHECK:STDERR: impl i32 as false {}
 // CHECK:STDERR:             ^~~~~
-// CHECK:STDERR: fail_impl_bad_interface.carbon:[[@LINE+3]]:13: note: type `bool` does not implement interface `ImplicitAs` [MissingImplInMemberAccessNote]
+// CHECK:STDERR: fail_impl_as_false.carbon:[[@LINE+4]]:13: note: type `bool` does not implement interface `ImplicitAs` [MissingImplInMemberAccessNote]
 // CHECK:STDERR: impl i32 as false {}
 // CHECK:STDERR:             ^~~~~
+// CHECK:STDERR:
 impl i32 as false {}
 
-// CHECK:STDOUT: --- fail_impl_bad_interface.carbon
+// --- fail_impl_as_type.carbon
+
+library "[[@TEST_NAME]]";
+
+// CHECK:STDERR: fail_impl_as_type.carbon:[[@LINE+4]]:1: error: impl as non-facet-type [ImplAsNonFacetType]
+// CHECK:STDERR: impl bool as type {}
+// CHECK:STDERR: ^~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
+impl bool as type {}
+
+// --- fail_impl_as_type_where.carbon
```

Comment:
We often seem to name files `fail_todo_...` or `todo_fail...` when the failure is expected, but a bug, and `file_test` has support for that.

Reply:
I don't honestly expect this particular test to pass even when the code is improved to handle more than one interface, since "impl as <something with no interface requirements>" should likely not be allowed in general. I added a comment.